pkg/splunk/enterprise/clustermaster.go

```diff
@@ -58,9 +58,6 @@ func ApplyClusterMaster(client splcommon.ControllerClient, cr *enterprisev1.Clus
 	if !reflect.DeepEqual(cr.Status.SmartStore, cr.Spec.SmartStore) ||
 		AreRemoteVolumeKeysChanged(client, cr, SplunkClusterMaster, &cr.Spec.SmartStore, cr.Status.ResourceRevMap, &err) {
 
-		if err != nil {
-			return result, err
-		}
```

Comment:
Any reason we are removing this error check? This was introduced to handle the case when somebody accidentally deletes the secret object consisting the S4 keys.

Reply:
I knew this would come up :). If you look at the code for `AreRemoteVolumeKeysChanged`, there is no case in that function where we are returning `true` and also returning `err` as well.

Reply:
Good point Gaurav. That was a miss i guess. In that case, we need to handle it differently such that we still return for that case.

Reply:
yes, you are right! we should error out in that case. So in short, we need to take care of case where it returns false with error. Let me fix it and push a change again.